.github/workflows/test_release.yaml

```diff
@@ -31,58 +49,114 @@ jobs:
-         - {version: '1.17', kind: "v0.11.1", image: 'kindest/node:v1.17.17@sha256:66f1d0d91a88b8a001811e2f1054af60eef3b669a9a74f9b6db871f2f1eeed00', crd_version: 'v1' }
-         - {version: '1.18', kind: "v0.11.1", image: 'kindest/node:v1.18.19@sha256:7af1492e19b3192a79f606e43c35fb741e520d195f96399284515f077b3b622c', crd_version: 'v1' }
-         - {version: '1.19', kind: "v0.11.1", image: 'kindest/node:v1.19.11@sha256:07db187ae84b4b7de440a73886f008cf903fcf5764ba8106a9fd5243d6f32729', crd_version: 'v1' }
-         - {version: '1.20', kind: "v0.11.1", image: 'kindest/node:v1.20.7@sha256:cbeaf907fc78ac97ce7b625e4bf0de16e3ea725daf6b04f930bd14c67c671ff9', crd_version: 'v1' }
-         - {version: '1.21', kind: "v0.11.1", image: 'kindest/node:v1.21.1@sha256:69860bda5563ac81e3c0057d654b5253219618a22ec3a346306239bba8cfa1a6', crd_version: 'v1' }
-         - {version: '1.22', kind: "v0.11.1", image: 'kindest/node:v1.22.0@sha256:b8bda84bb3a190e6e028b1760d277454a72267a5454b57db34437c34a588d047', crd_version: 'v1' }
-         - {version: '1.23', kind: "v0.11.1", image: 'kindest/node:v1.23.0@sha256:49824ab1727c04e56a21a5d8372a402fcd32ea51ac96a2706a12af38934f81ac', crd_version: 'v1' }
-    name: k8s ${{ matrix.configuration.version }}
+      matrix: ${{fromJson(needs.setmatrix.outputs.matrix)}}
+    name: e2e test k8s ${{ matrix.configuration.version }}
     steps:
-    - name: checkout
+    - name: Checkout
       uses: actions/checkout@v3
       with:
         fetch-depth: '5'
-    - name: build local Image
+
+    - name: Build local image
       run: docker build -t metacontrollerio/metacontroller:dev .
-    - name: setup kind
+
+    - name: Setup kind
       uses: helm/kind-action@v1.2.0
       with:
         version: ${{ matrix.configuration.kind }}
         node_image: ${{ matrix.configuration.image }}
         cluster_name: kind
         wait: 120s
+
     - name: Load metacontroller image
       run: |
         kind load docker-image metacontrollerio/metacontroller:dev
-    - name: Replace CRD mnifests to v1beta1
+
+    - name: Replace CRD manifests to v1beta1
       if: ${{ matrix.configuration.crd_version == 'v1beta1' }}
       run: |
         sed -i 's/crds-v1/crds-v1beta1/' manifests/production/kustomization.yaml
         for file in examples/**/*.yaml; do sed -i 's/apiextensions.k8s.io\/v1/apiextensions.k8s.io\/v1beta1/g' $file; done
+
     - name: Apply metacontroller manifests
       run: |
         kubectl apply -k manifests/dev
         sleep 60s
         kubectl get pods -n metacontroller
+
     - name: Run e2e tests
       timeout-minutes: 30
       run: |
         cd examples
         ./test.sh --crd_version ${{ matrix.configuration.crd_version }}
+
+  helmtests:
+    runs-on: ubuntu-20.04
+    needs: ['setmatrix']
+    strategy:
+      fail-fast: false
+      # TODO: ask about skipping CRD versions v1beta because it is not in helm chart
```

Comment:
Hi @grzesuav , I have added the github action helm chart testing for this PR. This will check for changes in the chart. If changes are found, deploy a kind cluster and run helm lint/install. It will test all the value files under the deploy/helm/ci directory. I created a symlink for default_values.yaml to values.yaml to make sure the default values are tested. One thing I noticed, is currently the helm chart only includes the v1 CRD and not the previous v1beta CRD. As a result, I removed the v1beta version for kind clusters from this test.

I am thinking either we deprecate the older version for k8s or update the charts in a manner to support both (outside scope of this ticket). Thoughts?

Please let me know if you have any feedback. I will remove from draft to ready for review once this TODO is addressed.

Reply:
hi, actually we can always run this test, as always at least metacontroller version in helm will change. Removing v1beta1 is ok, I need to fix other PR and set minimal version to 1.16, but will do it as breaking change

Reply:
I don't think you are able to template in the crds directory.
As far as I know this is only possible in the templates directory.

Reply:
@roelvdberg  Yep you are correct, I gave it a shot and no luck. Helm docs say the same. I started to look into using subcharts for the different CRD versions but it became unnecessarily complex. I will continue with ignoring k8s versions that do no support v1 in this PR because it does not break the existing setup. Thanks @grzesuav and @roelvdberg  for feedback